assets/js/main.js

```diff
@@ -147,75 +147,97 @@ $('.progress-bar').each(function(index) {
-      this.value=Math.max(this.value,minThumb.value-(-2));
-      var totalSteps = parseInt(this.max)-parseInt(this.min);
-      var percent=(100/totalSteps)*parseInt(this.value)-(100/totalSteps)*parseInt(this.min);
-      var $container = $(this).parent();
-      $container.find("div[inverse-right]").css("width",(100 - percent) +'%');
-      $container.find("div[range]:last").css("right",(100-percent)+'%')
-      $container.find("div[sign-two],span[thumb]:last").css("left", percent+'%');
-      $container.find("div[sign-two] span").text(this.value + label);
-    });
-  };
+  $(element2).on("input", function () {
+    var label = $(this).attr("data-label");
+    var minThumb = $(this).parent().find(".thumb-1")[0];
+    this.value = Math.max(this.value, minThumb.value - (-2));
+    var totalSteps = parseInt(this.max) - parseInt(this.min);
+    var percent = (100 / totalSteps) * parseInt(this.value) - (100 / totalSteps) * parseInt(this.min);
+    var $container = $(this).parent();
+    $container.find("div[inverse-right]").css("width", (100 - percent) + '%');
+    $container.find("div[range]:last").css("right", (100 - percent) + '%')
+    $container.find("div[sign-two],span[thumb]:last").css("left", percent + '%');
+    $container.find("div[sign-two] span").text(this.value + label);
+  });
+};
 
-  $('.dp-show-tips').on('click', function(){
-    $(this).closest('.field-item').next('.dp-advice').slideToggle();
+$('.dp-show-tips').on('click', function () {
+  $(this).closest('.field-item').next('.dp-advice').slideToggle();
 
-    var $showPanelLink = $(this).find('.dp-show-text');
-    if ($(this).find('.icon-close').hasClass('rotation')) {
-      $showPanelLink.html($(this).attr('data-text-hide'));
-      } else {
-        $showPanelLink.html($(this).attr('data-text-show'));
-    }
-    $(this).find('.icon-close').toggleClass('rotation');
+  var $showPanelLink = $(this).find('.dp-show-text');
+  if ($(this).find('.icon-close').hasClass('rotation')) {
+    $showPanelLink.html($(this).attr('data-text-hide'));
+  } else {
+    $showPanelLink.html($(this).attr('data-text-show'));
+  }
+  $(this).find('.icon-close').toggleClass('rotation');
 
-  });
+});
 
+// Carousel
+
+(function () {
+  $(document).ready(function () {
+    return $("input[type='radio']").change(function () {
+      var $current_radiobutton, current_panel_id, current_radiobutton_name;
+      $current_radiobutton = $(this);
+      current_radiobutton_name = $current_radiobutton.attr('name');
+      current_panel_id = `#${$current_radiobutton.attr('value')}_dpslide`;
+      return $(`[name='${current_radiobutton_name}']`).each((i, element) => {
+        var $panel, $radiobutton, panel_id;
+        $radiobutton = $(element);
+        panel_id = `#${$radiobutton.attr('id')}_dpslide`;
+        $panel = $(panel_id);
+        if (panel_id === current_panel_id) {
+          return $panel.show();
+        } else {
+          return $panel.hide();
+        }
+      });
```

Comment:
Why not add the same name for each slide and radios, so you can target them directly?
```
<div class="dp-carousel-slide" id="primero_dpslide" name="carousel1">
  ...
</div>
<div class="dp-carousel-slide" id="segundo_dpslide" name="carousel1">
  ...
</div>
```
So then target al the slides to hide
```suggestion
      var slideId = $(this).prop("id");
      var slidesName = $(this).prop("name");
      // hide all slides
      $('div[name="'+slidesName+'"]').hide();
      // show the one with the correct id
      $(slideId+ '_slide').show();
      });
```
Am I understanding this right?

Reply:
I tried that and it doesn't work

Reply:
I will try directly on the code, the logic is sound I think 🤔

Reply:
```
      var slideId = $(this).prop("id");
      var slidesName = $(this).prop("name");
      // hide all slides
      $('div[name="'+slidesName+'"]').hide();
      // show the one with the correct id
      $('#'+slideId+ '_dpslide').show();
```
You can try it this way and it works, I've tested just now. 
Also there is no need to do a return for each event, with just executing the code it's ok. You could replace the whole function by this:
```
// Carousel

  $(document).ready(function () {
    $("input[type='radio'].dp-carousel-dot").change(function () {
      var slideId = $(this).prop("id");
      var slidesName = $(this).prop("name");
      // hide all slides
      $('div[name="'+slidesName+'"]').hide();
      // show the one with the correct id
      $('#'+slideId+ '_dpslide').show();
    });
  });
```

Reply:
Great, thanks! I change it